front-end/src/components/helpers/IntentSearch.js

```diff
@@ -4,12 +4,18 @@ export default class IntentSearch{
     /**
      * @param {Array} intents array of intents to search through
      * @param {String} searchString string to search by
-     * @return {Object} an array of intents filtered by searchString and a mapping of each intent to indices describing where the searchString appears in its question
+     * @param {Boolean} [starFilter=false] if needed to filter by star
+     * @param {Boolean} [flagFilter=false] if needed to filter by flag
+     * @return {Object} an array of intents filtered by searchString, starFilter, and flagFilter and a mapping of each intent to indices describing where the searchString appears in its question
      * */
-    filterIntents = (intents, searchString) => {
+    filterIntents = (intents, searchString,starFilter=false, flagFilter=false) => {
         let filteredIntents = [];
         let searchSlices = {};
-        intents.forEach((intent) => this.addFilteredIntent(intent, searchString, filteredIntents, searchSlices));
+        intents.forEach((intent) => {
```

Comment:
![image](https://user-images.githubusercontent.com/82854864/204433767-30dead6e-d082-41a0-b15d-1307697de78d.png)
I'm wondering, if we have both of the options toggled, don't we want only the questions with both to appear? I'm open to the interpretation of it including all star, flag, and double star-flagged though, nothing major, just a thought on how someone intuitively uses and views the feature.

Reply:
Yeah I was thinking of the filters more like a retail store where you check off multiple checkboxes to see options that apply to either instead of only both

Reply:
Y = AB vs Y = A+B strikes again!